javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -64,7 +64,7 @@ static boolean areEqual(Iterable<?> iterable1, Iterable<?> iterable2) {
     static <T, C, R extends Iterable<T>> Map<C, R> groupBy(Traversable<T> source, Function<? super T, ? extends C> classifier, Function<? super Iterable<T>, R> mapper) {
         Objects.requireNonNull(classifier, "classifier is null");
         Objects.requireNonNull(mapper, "mapper is null");
-        final java.util.Map<C, Collection<T>> mutableResults = new java.util.LinkedHashMap<>();
+        final java.util.Map<C, Collection<T>> mutableResults = new java.util.LinkedHashMap<>(source.size());
```

Comment:
or I could change it to `new java.util.LinkedHashMap<>(source.isTraversableAgain() ? source.size() : 16)`, whichever you prefer :)